src/ui/hook/useCopyToClipboard.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useState } from 'react'
+import { useTimeout } from '@/ui/hook/useTimeout'
+
+type HandleCopy = (textToCopy: string) => Promise<void>
+type CopyState = 'idle' | 'success' | 'error'
+
+export const useCopyToClipboard = (
+  copyResetDelay?: number,
+  handler?: (isCopied: boolean) => void
```

Comment:
If you have the copy state available, you can use it directly instead of relying on a callback. Otherwise, there is a mix of the reactive state style and the callback style, imho.